### PR TITLE
WRO-12375: Increase code coverage for TabLayout/TooltipDecorator

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Increase code coverage for TabLayout/TooltipDecorator.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
For the TabLayout/TabLayout.js I've only managed to increase coverage to 73.33%. Even though for the entire TabLayout folder the coverage is 84.05%, if the reviewers have any sugestions on how to do additional tests I would be grateful if they will share them in a comment below.

### Links
[//]: # (Related issues, references)
WRO-12375.

### CODE REVIEW

File: TooltipDecorator/tests/TooltipDecorator-specs.js [hunk 1]
```diff
@@ -23,4 +29,160 @@ describe('TooltipDecorator', () => {
...
+
+			act(() => button.blur());
+			fireEvent.mouseOut(button);
+
+			await waitFor(() => {
+				expect(screen.queryByText('Tooltip')).toBeNull();
+			});
+		});
+
+		// This test throws 'Warning: Each child in a list should have a unique "key" prop.'
```
Comment (daniel-stoian-lgp) [inline TooltipDecorator/tests/TooltipDecorator-specs.js:88]: try to mock the implementation of console.error to supress the warning and remove this comment

File: TooltipDecorator/tests/TooltipDecorator-specs.js [hunk 2]
```diff
@@ -23,4 +29,185 @@ describe('TooltipDecorator', () => {
...
+			act(() => button.focus());
+			fireEvent.mouseOver(button);
+
+			await waitFor(() => {
+				expect(screen.getByText('Tooltip')).not.toBeNull();
+			});
+		});
+
+		describe('Tooltip position', () => {
+			test('above', async () => {
```
Comment (daniel-stoian-lgp) [inline TooltipDecorator/tests/TooltipDecorator-specs.js:107]:
```suggestion
			test('should have "above" className when tooltipPosition is set to \'above\'', async () => {
```

Comment (daniel-stoian-lgp) [reply TooltipDecorator/tests/TooltipDecorator-specs.js:107]: please do the same for all the following tests

File: TooltipDecorator/tests/TooltipDecorator-specs.js (hunk 1 above)
Comment (alexandrumorariu) [reply TooltipDecorator/tests/TooltipDecorator-specs.js]: Done

File: TooltipDecorator/tests/TooltipDecorator-specs.js [hunk 3]
```diff
@@ -23,4 +29,185 @@ describe('TooltipDecorator', () => {
...
+			act(() => button.blur());
+			fireEvent.mouseOut(button);
+
+			await waitFor(() => {
+				expect(screen.queryByText('Tooltip')).toBeNull();
+			});
+		});
+
+		test('should render a tooltip if hovered for \'tooltipRelative\'', async () => {
+			console.error = jest.fn();	// eslint-disable-line no-console
```
Comment (daniel-stoian-lgp) [inline TooltipDecorator/tests/TooltipDecorator-specs.js:89]:
```suggestion
		// eslint-disable-next-line no-console
		console.error.mockImplementation();
```

File: TooltipDecorator/tests/TooltipDecorator-specs.js [hunk 4]
```diff
@@ -1,8 +1,14 @@
+import {FloatingLayerDecorator} from '@enact/ui/FloatingLayer';
 import '@testing-library/jest-dom';
-import {render, screen} from '@testing-library/react';
+import {act, fireEvent, render, screen, waitFor} from '@testing-library/react';
 
+import Button from '../../Button';
+import TooltipDecorator from '../TooltipDecorator';
```
Comment (MikyungKim) [inline TooltipDecorator/tests/TooltipDecorator-specs.js:6]:
```suggestion
import Button from '../../Button';

import TooltipDecorator from '../TooltipDecorator';
```

File: TooltipDecorator/tests/TooltipDecorator-specs.js [hunk 5]
```diff
@@ -23,4 +29,185 @@ describe('TooltipDecorator', () => {
...
+					<TooltipButton tooltipDelay={0} tooltipText={tooltipText}>Label</TooltipButton>
+				</FloatingLayerController>
+			);
+
+			const button = screen.getByRole('button');
+			act(() => button.focus());
+			fireEvent.mouseOver(button);
+
+			await waitFor(() => {
+				expect(screen.getByText('Tooltip')).not.toBeNull();
```
Comment (MikyungKim) [inline TooltipDecorator/tests/TooltipDecorator-specs.js:60]:
```suggestion
				expect(screen.getByText('Tooltip')).toBeInTheDocument();
```

File: TooltipDecorator/tests/TooltipDecorator-specs.js [hunk 6]
```diff
@@ -23,4 +29,185 @@ describe('TooltipDecorator', () => {
...
+			});
+
+			act(() => button.blur());
+			fireEvent.mouseOut(button);
+
+			await waitFor(() => {
+				expect(screen.queryByText('Tooltip')).toBeNull();
```
Comment (MikyungKim) [inline TooltipDecorator/tests/TooltipDecorator-specs.js:84]:
```suggestion
				expect(screen.queryByText('Tooltip')).not.toBeInTheDocument();
```

File: TooltipDecorator/tests/TooltipDecorator-specs.js [hunk 7]
```diff
@@ -23,4 +29,185 @@ describe('TooltipDecorator', () => {
...
+					<TooltipButton tooltipDelay={0} tooltipRelative tooltipText={tooltipText}>Label</TooltipButton>
+				</FloatingLayerController>
+			);
+
+			const button = screen.getByRole('button');
+			act(() => button.focus());
+			fireEvent.mouseOver(button);
+
+			await waitFor(() => {
+				expect(screen.getByText('Tooltip')).not.toBeNull();
```
Comment (MikyungKim) [inline TooltipDecorator/tests/TooltipDecorator-specs.js:102]:
```suggestion
				expect(screen.getByText('Tooltip')).toBeInTheDocument();
```

**Review by MikyungKim — COMMENTED** [review]
Some of the uncovered code from TabLayout is for the behavior of PopupTabLayout, but it looks hard to hit the code since it needs the real dom node for most cases.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**